### PR TITLE
vCow contracts

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Added vCow related stuff 

- ABI
- Generated types
- Temporary Rinkeby address
- Hook

  # To Test

No testing possible yet

### CODE REVIEW

File: src/custom/abis/types/factories/GPv2Settlement__factory.ts [hunk 1]
```diff
@@ -2,86 +2,92 @@
...
 
-import { Contract, Signer, utils } from 'ethers'
-import { Provider } from '@ethersproject/providers'
-import type { GPv2Settlement, GPv2SettlementInterface } from '../GPv2Settlement'
+import { Contract, Signer, utils } from "ethers";
+import { Provider } from "@ethersproject/providers";
+import type {
+  GPv2Settlement,
+  GPv2SettlementInterface,
+} from "../GPv2Settlement";
```
Comment (W3stside) [inline src/custom/abis/types/factories/GPv2Settlement__factory.ts:10]: whats up with the semi colons?

Comment (alfetopito) [reply src/custom/abis/types/factories/GPv2Settlement__factory.ts:10]: Auto generated code, will revert this file

**Comment by W3stside** [comment]
@alfetopito does the auto-generated type stuff have auto compile in package json? where is it coming from

**Review by W3stside — APPROVED** [review]

**Comment by alfetopito** [comment]
> @alfetopito does the auto-generated type stuff have auto compile in package json? where is it coming from

No, because the default path is for Uniswap contracts.
I manually changed it to `/custom` to generate them => https://github.com/gnosis/cowswap/blob/claim/package.json#L159

I don't think we need to re-generate them every time and also it would overwrite some custom changes we need, namely exporting the originally generated exports.

**Pull request merged**